workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+---
+name: 'Reddit Viewer'
+description: 'Use Flask and the PRAW library to make a subreddit browser.'
+author: '@KaiDevrim'
+---
+
+![The end result](https://cloud-fz3kdxk80.vercel.app/image.png)
+
+Python combined with Flask is an amazing way to understand how to make a website backend. Understanding how to make a backend is very simple and in this tutorial, we will be making our own Reddit web client that fetches directly from the reddit API.
+
+# Step 1: Setup
+
+Since this is just Python there isn't much setup. Just go to [repl.it](https://repl.it) and make a new repl. Make sure you import the 2 libraries required for this, and 2 for keeping your api key safe.
+
+```py
+import praw
+from dotenv import load_dotenv
+import os
+from flask import Flask, render_template, request
+
+```
+
+In the same folder as your main.py, make a file called `.env`. This is where we will store the api keys. Also go back to the main.py file and add these four lines to load your env file and flask,
+
+```py
+load_dotenv()
+app = Flask(__name__)
+
+if __name__ == '__main__':
+  app.run(host='0.0.0.0' , port = 9060)
+```
+
+The first one just finds and loads the your .env file and the second one says that this python file is the main file to run flask in. The third line is just doing a simple if check that will then run flask on localhost with the port being 9060.
+
+# Step 2: Getting Reddit API Key
+
+Getting your api key is very simple, go to the [Reddit Apps page](https://www.reddit.com/prefs/apps/) and click on `create another app...` at the bottom. ![The create another app button at the bottom](https://cloud-i2087nmyc.vercel.app/image.png)
+
+You should have a little section with `name`, `type`, `description`, `about url`, `redirect uri`. We only need the `name`, `type`, and `redirect uri`. For the name fill it with any name you want. For the type choose `script` and for the redirect uri, this doesn't matter, so just put `http://www.example.com/unused/redirect/uri`. ![The finished application creation](https://cloud-f67e393ze.vercel.app/image.png)
+
+In your repl.it open your .env file and make a variable called `SECRET` and another one called `ID`. Now go back to your reddit apps section and copy the secret from the secret part to your SECRET variable in your env file. Also copy the section just under your personal use script to your ID variable.
```

Comment:
Add some more steps on how to add variables to the .env variable. maybe even add a code block that demonstrates. The way I would structure this paragraph is:

In the Apps page on Reddit that you were on before, copy the API secret (image showing where to copy it from, with your secret blocked out). Then, head back over to your repl. In the `.env` file, create a variable called SECRET, like so:

```
SECRET=YOUR_API_SECRET
```

replacing `YOUR_API_SECRET` with the key you copied.

Repeat the same step for ID

Reply:
Done!